src/stores/messages/MessagesDataProviderStore.ts

```diff
@@ -328,37 +321,22 @@ export default class MessagesDataProviderStore implements MessagesDataStore {
 	public onNextChannelResponse = (messages: EventMessage[]) => {
 		this.lastNextChannelResponseTimestamp = null;
 
-		// eslint-disable-next-line no-param-reassign
-		messages = messages.filter(
-			msg => msg.messageId !== this.messagesStore.selectedMessageId?.valueOf(),
-		);
-
-		const prevMessages =
-			this.messages.length > 0
-				? messages
-						.filter(
-							message =>
-								timestampToNumber(message.timestamp) <
-								timestampToNumber(this.messages[0].timestamp),
-						)
-						.sort((a, b) => timestampToNumber(b.timestamp) - timestampToNumber(a.timestamp))
-				: [];
-
 		const nextMessages = messages
-			.slice(0, messages.length - prevMessages.length)
+			.filter(message => !this.messages.map(msg => msg.messageId).includes(message.messageId))
 			.sort((a, b) => timestampToNumber(b.timestamp) - timestampToNumber(a.timestamp));
```

Comment:
i dont think sorting is necessary

Reply:
I also have a question about this line from master: 
			this.messages = [...new Map(newMessagesList.map(item => [item.messageId, item])).values()];
It looks weird, should I get it back or just newMessagesList is enough?

Reply:
ye, im not sure if Map.values() guarantees same order
i guess it was made for filtering duplicates, so it could be filtered the same way you will filter it in onNextChannelResponse